# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.cpp
# Commit: 01c2bc6d727a
# Full Hash: 01c2bc6d727a70be74dbf7babb7f6294824a571a
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 10 - CookieStore API - CHIP implementation shared between Document and CookieStore, r=cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220288
# ==============================================================================

diff -r 2d925ea3da19 -r 01c2bc6d727a netwerk/cookie/CookieCommons.cpp
--- a/netwerk/cookie/CookieCommons.cpp	Tue Sep 10 16:31:41 2024 +0000
+++ b/netwerk/cookie/CookieCommons.cpp	Tue Sep 10 16:31:41 2024 +0000
@@ -508,30 +508,41 @@
 }
 
 // static
-bool CookieCommons::ShouldIncludeCrossSiteCookieForDocument(
-    Cookie* aCookie, dom::Document* aDocument) {
+bool CookieCommons::ShouldIncludeCrossSiteCookie(Cookie* aCookie,
+                                                 bool aPartitionForeign,
+                                                 bool aInPrivateBrowsing,
+                                                 bool aUsingStorageAccess) {
   MOZ_ASSERT(aCookie);
-  MOZ_ASSERT(aDocument);
 
   int32_t sameSiteAttr = 0;
   aCookie->GetSameSite(&sameSiteAttr);
 
+  return ShouldIncludeCrossSiteCookie(
+      sameSiteAttr, aCookie->IsPartitioned() && aCookie->RawIsPartitioned(),
+      aPartitionForeign, aInPrivateBrowsing, aUsingStorageAccess);
+}
+
+// static
+bool CookieCommons::ShouldIncludeCrossSiteCookie(int32_t aSameSiteAttr,
+                                                 bool aCookiePartitioned,
+                                                 bool aPartitionForeign,
+                                                 bool aInPrivateBrowsing,
+                                                 bool aUsingStorageAccess) {
   // CHIPS - If a third-party has storage access it can access both it's
   // partitioned and unpartitioned cookie jars, else its cookies are blocked.
   //
   // Note that we will only include partitioned cookies that have "partitioned"
   // attribution if we enable opt-in partitioning.
-  if (aDocument->CookieJarSettings()->GetPartitionForeign() &&
+  if (aPartitionForeign &&
       (StaticPrefs::network_cookie_cookieBehavior_optInPartitioning() ||
-       (aDocument->IsInPrivateBrowsing() &&
+       (aInPrivateBrowsing &&
         StaticPrefs::
             network_cookie_cookieBehavior_optInPartitioning_pbmode())) &&
-      !(aCookie->IsPartitioned() && aCookie->RawIsPartitioned()) &&
-      !aDocument->UsingStorageAccess()) {
+      !aCookiePartitioned && !aUsingStorageAccess) {
     return false;
   }
 
-  return sameSiteAttr == nsICookie::SAMESITE_NONE;
+  return aSameSiteAttr == nsICookie::SAMESITE_NONE;
 }
 
 bool CookieCommons::IsSafeTopLevelNav(nsIChannel* aChannel) {